flink-yarn/src/main/java/org/apache/flink/yarn/YarnApplicationFileUploader.java

```diff
@@ -300,19 +310,22 @@ public FileVisitResult visitFile(java.nio.file.Path file, BasicFileAttributes at
 
 		final ArrayList<String> classPaths = new ArrayList<>();
 		providedSharedLibs.forEach(
-			(fileName, fileStatus) -> {
-				localResources.put(
-					fileName,
-					Utils.registerLocalResource(
-						fileStatus.getPath(),
-						fileStatus.getLen(),
-						fileStatus.getModificationTime(),
-						LocalResourceVisibility.PUBLIC));
-
-				if (!isFlinkDistJar(fileName)) {
-					classPaths.add(fileName);
-				}
-			});
+				(fileName, fileStatus) -> {
+					final Path filePath = fileStatus.getPath();
+					LOG.debug("Using remote file {} to register local resource", filePath);
+
+					final YarnLocalResourceDescriptor descriptor = YarnLocalResourceDescriptor
+							.fromFileStatus(fileName, fileStatus, LocalResourceVisibility.PUBLIC);
+					localResources.put(fileName, descriptor.toLocalResource());
+					remotePaths.add(filePath);
+					envShipResourceList.add(descriptor);
+
+					if (!isFlinkDistJar(fileName) && !isPlugin(fileStatus.getPath())) {
```

Comment:
I think we could not use `fileName` to check `isFlinkDistJar ` since it is relative path, not the real file name(e.g. lib/flink-dist_2.11-1.11-SNAPSHOT.jar). So i suggest to use the `key` or `relative` to instead of `fileName` to avoid confusing.

`fileStatus.getPath()` could be replaced with `filePath`.

Reply:
You mean use the `filePath.getName()`?

Reply:
Yes, i think it could not work here.

```
	private static boolean isFlinkDistJar(String fileName) {
		return fileName.startsWith("flink-dist") && fileName.endsWith("jar");
	}
```

Reply:
yes, you are right, I already updated it.